internal/clientconn/listener.go

```diff
@@ -108,7 +107,12 @@ func (l *Listener) Run(ctx context.Context) error {
 		go func() {
 			connID := fmt.Sprintf("%s -> %s", netConn.RemoteAddr(), netConn.LocalAddr())
 
-			runCtx, runCancel := ctxutil.WithDelay(ctx.Done(), delay)
+			// give clients a few seconds to disconnect after ctx is canceled
+			runCancelDelay := l.opts.TestRunCancelDelay
```

Comment:
I don't understand this part. What's the difference between this and `context.WithTImeout`?